BlueprintUI/Sources/Layout/Decorate.swift

```diff
@@ -0,0 +1,236 @@
+//
+//  Decorate.swift
+//  BlueprintUI
+//
+//  Created by Kyle Van Essen on 11/4/20.
+//
+
+import UIKit
+
+
+///
+/// Places a decoration element behind or in front of the given `wrapped` element,
+/// and positions it according to the `position` parameter.
+///
+/// The size and position of the element is determined only by the `wrapped`
+/// element, the `decoration` element does not affect the layout at all.
+///
+/// Example
+/// -------
+/// The arrows represent the measured size of the element for layout purposes.
+/// ```
+/// ┌───────────────────┐     ┌──────┐
+/// │    Decoration     │     │      │
+/// │ ┏━━━━━━━━━━━━━━━┓ │ ▲   │      ┣━━━━━━━━━━┓   ▲
+/// │ ┃               ┃ │ │   └─┳────┘          ┃   │
+/// │ ┃    Wrapped    ┃ │ │     ┃    Wrapped    ┃   │
+/// │ ┃               ┃ │ │     ┃               ┃   │
+/// │ ┗━━━━━━━━━━━━━━━┛ │ ▼     ┗━━━━━━━━━━━━━━━┛   ▼
+/// └───────────────────┘
+///   ◀───────────────▶         ◀───────────────▶
+/// ```
+public struct Decorate : ProxyElement {
+    
+    /// The element which provides the sizing and measurement.
+    /// The sizing and position of the `Decorate` element is determined
+    /// by this element.
+    public var wrapped : Element
+    
+    /// The element which is used to draw the decoration.
+    /// It does not affect sizing or positioning.
+    public var decoration : Element
+    
+    /// Where the decoration should be positioned in the z-axis: Above or below the wrapped element.
+    public var layering : Layering
+    
+    /// How the `decoration` should be positioned in respect to the `wrapped` element.
+    public var position : Position
+    
+    /// Creates a new instance with the provided overflow, background, and wrapped element.
+    public init(
+        layering : Layering,
+        position: Position,
+        wrapping: () -> Element,
```

Comment:
I like the closure since it gives you an immediate scope to set things up, but I'll remove for consistency for now.

Reply:
I think I'll keep it on the `Element` extension though since it makes the code read a bit more nicely, eg:

```
        .thingy()
        .decorate(layering: .below, position: .inset(5)) {
            Box(backgroundColor: .init(white: 0.0, alpha: 0.1), cornerStyle: .rounded(radius: 17))
        }
```